### DISCUSSION TITLE
too many open files in system (os error 23)

### DISCUSSION THREAD

**Discussion by mountingJin** [geldata/gel#2466] [Q&A] [answered]
Hi. Thanks for making a cool db.
I am a beginner.

I went into <edgedb -I test> and practiced edbeql, but after several times I got too many open files in system (os error 23) error.

Even if it ends well with \q, it's the same.
  It's the same until I kill it with kill -9 PID.

Looking at lsof, it keeps rising every time the postgres process hits a query or two.

Mac OSX bir sur: 11.2.3

I can't even practice querying the tutorial.
How do I do it?

**Comment by elprans** [comment, marked as answer]
Hi @mountingJin.  Could you post a full error traceback or a screenshot please?

**Comment by mountingJin** [comment]
https://github.com/mountingJin/image/blob/main/chrome%20error.txt

this is chrome error when open chrome.



https://github.com/mountingJin/image/blob/main/instance.png

It is an error in the <edgedb -I dracula> shell.



And other programs and webstorm editors don't open on properly, and clients sometimes get [ENFILE] something proxy error.
It's just an error I hit SELECT PERSON 40 to 50 times in Shell. I don't think it's a problem because I wrote the wrong code.
Let me know if you need anything else.

**Reply by elprans** [reply]
You might want to increase the kernel file limit.  It's actually quite low on macOS by default.  See [this StackExchange answer](https://superuser.com/questions/433746/is-there-a-fix-for-the-too-many-open-files-in-system-error-on-os-x-10-7-1#answer-443168) on how to do this.

**Reply by mountingJin** [reply]
thank you. It's working well.
But is it correct that the postgres process increases every time hit the query? When are the processes closed?
When uploading to a web server, do I just have to put the limit of files in the same way?

**Reply by elprans** [reply]
EdgeDB maintains a pool of PostgreSQL connections.  In the current release the upper limit for the maximum number of connections is hardcoded to 100, but the development version, (which will become Beta 2 next week), now adjusts that based on the amount of available system resources.

**Reply by mountingJin** [reply]
thank you so much @elprans , Now I understand about pool.
